source/game/field/obj/ObjectShip64.cc

```diff
@@ -0,0 +1,83 @@
+#include "ObjectShip64.hh"
+#include <egg/math/Matrix.hh>
+#include <egg/math/Vector.hh>
+#include <game/field/ObjectCollisionCylinder.hh>
+#include <game/field/RailInterpolator.hh>
+#include <game/field/obj/ObjectBase.hh>
+#include <game/field/obj/ObjectCollidable.hh>
+
+namespace Field {
+
+ObjectShip64::ObjectShip64(const System::MapdataGeoObj &params) : ObjectCollidable(params) {}
+
+void ObjectShip64::init() {
+    m_railInterpolator->init(0, 0);
+
+    m_tangent = m_railInterpolator->curTangentDir();
+    m_railInterpolator->setPerPointVelocities(true);
+
+    calc();
+}
+
+void ObjectShip64::calc() {
+    m_railInterpolator->calc();
+
+    setPos(m_railInterpolator->curPos());
+
+    fprintf(stderr, "m_pos: %f, %f, %f\n", m_pos.x, m_pos.y, m_pos.z);
+
+    EGG::Vector3f v;
+    v = m_tangent - m_railInterpolator->curTangentDir();
+    v *= 0.2;
+    v += m_tangent;
+    m_tangent = v;
+    m_tangent.normalise();
+
+    setMatrixFromOrthonormalBasisAndPos(m_tangent);
+}
+
+const char *ObjectShip64::getKclName() const {
+    return "DKShip64";
+}
+
+void ObjectShip64::createCollision() {
+    ObjectCollidable::createCollision();
+
+    constexpr f32 RADIUS = 1500.0, HEIGHT = 3500.0;
+
+    m_auxCollision = new ObjectCollisionCylinder(RADIUS, HEIGHT, collisionCenter());
+}
+
+void ObjectShip64::calcCollisionTransform() {
+    ObjectCollidable::calcCollisionTransform();
```

Comment:
Missing a `calcTransform()` call. (It's `UpdateMatrix` in Ghidra)
```suggestion
    ObjectCollidable::calcCollisionTransform();
    calcTransform();
```